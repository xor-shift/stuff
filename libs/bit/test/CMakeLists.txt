cmake_minimum_required(VERSION 3.25)

project(stuff_bit_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        bit.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::bit gtest gtest_main)
