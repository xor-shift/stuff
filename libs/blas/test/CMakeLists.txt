cmake_minimum_required(VERSION 3.25)

project(stuff_blas_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        matrix.cpp
        vector.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::blas gtest gtest_main)
