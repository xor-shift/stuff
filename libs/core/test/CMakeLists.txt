cmake_minimum_required(VERSION 3.25)

project(stuff_core_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        core.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::core gtest gtest_main)
