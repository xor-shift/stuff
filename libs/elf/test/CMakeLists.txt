cmake_minimum_required(VERSION 3.25)

project(stuff_elf_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        elf.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::elf gtest gtest_main)