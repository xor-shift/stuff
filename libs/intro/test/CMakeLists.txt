cmake_minimum_required(VERSION 3.25)

project(stuff_intro_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        function.cpp
        intro.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::intro gtest gtest_main)
target_compile_options(${PROJECT_NAME} PRIVATE -ftemplate-backtrace-limit=128)
