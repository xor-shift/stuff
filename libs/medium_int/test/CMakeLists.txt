cmake_minimum_required(VERSION 3.25)

project(stuff_medium_int_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        random.cpp

        carry.cpp
        tabled.cpp
        tabled_arithmetic.cpp
        random_arithmetic.cpp
        tabled_roundtrip.cpp
        random_roundtrip.cpp
)

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::medium_int stf::random gtest gtest_main)
