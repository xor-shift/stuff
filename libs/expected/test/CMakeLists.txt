cmake_minimum_required(VERSION 3.25)

project(stuff_expected_tests LANGUAGES CXX)

add_executable(${PROJECT_NAME}
        expected.cpp
        )

sanitize(${PROJECT_NAME})

target_link_libraries(${PROJECT_NAME} stf::expected gtest gtest_main)
